modules/zinit/zinit.go

```diff
@@ -7,38 +7,32 @@ import (
 	"net"
 	"strconv"
 	"strings"
+
+	"github.com/pkg/errors"
 )
 
 const defaultSocketPath = "/var/run/zinit.sock"
 
 // Client is a client for zinit action
 // it talks to zinit directly over its unis socket
 type Client struct {
-	socket string //path to the unix socket
-	conn   net.Conn
+	conn net.Conn
+	scan *bufio.Scanner
 }
 
 // New create a new Zinit client
-func New(socket string) *Client {
+func New(socket string) (*Client, error) {
```

Comment:
I'm trying to have all the constructor to never return an error.
That's why I had a `New` that just builds the object and then a `Connect` that dials and can return an error.

Reply:
I am not sure about this, it feels like a weird pattern that i have never encounter before. Also it just doubles the work you have to do to get a working client. IMHO if you always have to do New then Connect every time u need a client, you can always have a util method `New` or `Connect` that do that for u. No ?